From 63e5924b87164b02e9ac636ba24841ce5be992d2 Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Tue, 21 Jun 2022 23:03:04 +0000
Subject: [PATCH] disable using diagnose_if

---
 tensorflow/compiler/xla/util.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/tensorflow/compiler/xla/util.h b/tensorflow/compiler/xla/util.h
index 09cc7d6063e..1d4cd7c386f 100644
--- a/tensorflow/compiler/xla/util.h
+++ b/tensorflow/compiler/xla/util.h
@@ -433,18 +433,11 @@ void LogLines(int sev, absl::string_view text, const char* fname, int lineno);
 //
 //   int compute_absolute_value(int c)
 //     XLA_DIAGNOSE_ERROR_IF(c >= 0, "'c' is already positive.");
-#if ABSL_HAVE_ATTRIBUTE(diagnose_if)
-#define XLA_DIAGNOSE_ERROR_IF(...) \
-  __attribute__((diagnose_if(__VA_ARGS__, "error")))
-#else
 #define XLA_DIAGNOSE_ERROR_IF(...)
-#endif

 constexpr bool IsRuntimeEvaluated() {
 #ifdef __cpp_lib_is_constant_evaluated
   return !std::is_constant_evaluated();
-#elif ABSL_HAVE_BUILTIN(__builtin_is_constant_evaluated)
-  return !__builtin_is_constant_evaluated();
 #else
   return false;
 #endif
-- 
2.36.1

