From 5cf2c5cd8b3c8aafa7208b8e02f8b600efa1de3d Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 17 Nov 2023 14:34:29 +0100
Subject: [PATCH] Add ducc to static_deps

---
 tensorflow/BUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index 1583d8e6..cee5dd34 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -1457,7 +1457,7 @@ tf_cc_shared_library(
         "//tensorflow/core/distributed_runtime/rpc:grpc_session",
     ]),
     soversion = VERSION,
-    static_deps = PACKAGE_STATIC_DEPS,
+    static_deps = PACKAGE_STATIC_DEPS + ["@ducc//:__subpackages__"],
     visibility = ["//visibility:public"],
     win_def_file = ":tensorflow_filtered_def_file",
 )
-- 
2.39.3 (Apple Git-145)

