diff --git a/tensorflow/tsl/platform/default/build_config.bzl b/tensorflow/tsl/platform/default/build_config.bzl
--- tensorflow/tsl/platform/default/build_config.bzl
+++ tensorflow/tsl/platform/default/build_config.bzl
@@ -402,12 +402,9 @@
         # have grpc installed.
 
     genproto_deps = []
     for dep in deps:
-        if dep != "@com_google_protobuf//:protobuf_python":
-            genproto_deps.append(dep + "_genproto")
-        else:
-            genproto_deps.append("@com_google_protobuf//:well_known_types_py_pb2_genproto")
+        genproto_deps.append(dep + "_genproto")
 
     proto_gen(
         name = name + "_genproto",
         srcs = srcs,
