diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index b4a0c5dbce0..d62ac19a39e 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -89,7 +89,7 @@ REQUIRED_PACKAGES = [
     'keras_preprocessing ~= 1.1.2',
     'opt_einsum ~= 3.3.0',
     'protobuf >= 3.9.2',
-    'six ~= 1.15.0',
+    'six >= 1.15.0',
     'termcolor ~= 1.1.0',
     'typing_extensions ~= 3.7.4',
     'wheel ~= 0.35',

