From 714fb8709c4207a2226126a9c96143a37f7d256f Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 15 Sep 2023 11:06:27 +0200
Subject: [PATCH 07/12] Omit linking to layout_proto_cc if protobuf linkage is
 shared

---
 tensorflow/python/BUILD | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tensorflow/python/BUILD b/tensorflow/python/BUILD
index 9810f8ac..c7feacb6 100644
--- a/tensorflow/python/BUILD
+++ b/tensorflow/python/BUILD
@@ -1330,7 +1330,6 @@ tf_python_pybind_extension(
     features = ["-layering_check"],
     deps = [
         ":pywrap_densor_device_headers",
-        "//tensorflow/dtensor/proto:layout_proto_cc",
         "//tensorflow/python/lib/core:pybind11_lib",
         "//tensorflow/python/lib/core:pybind11_status_headers",
         "//third_party/python_runtime:headers",  # buildcleaner: keep
-- 
2.39.3 (Apple Git-145)

