From 6cc878513d44e08cd39b8f55f39f52b3c13893c8 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 27 Oct 2023 11:20:12 +0200
Subject: [PATCH 08/27] Add constraint to pybind11 systemlib

---
 third_party/systemlibs/pybind11.BUILD | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/third_party/systemlibs/pybind11.BUILD b/third_party/systemlibs/pybind11.BUILD
index 44e7496f..21ec073b 100644
--- a/third_party/systemlibs/pybind11.BUILD
+++ b/third_party/systemlibs/pybind11.BUILD
@@ -6,3 +6,9 @@ cc_library(
         "@local_tsl//third_party/python_runtime:headers",
     ],
 )
+
+# Needed by pybind11_bazel.
+config_setting(
+    name = "osx",
+    constraint_values = ["@platforms//os:osx"],
+)
