From 70bda39b9de02070156308d9e4070c9d0e651c38 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 15 Nov 2023 10:32:41 +0000
Subject: [PATCH 12/27] Add protobuf toolchain

---
 third_party/systemlibs/protobuf.BUILD | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/third_party/systemlibs/protobuf.BUILD b/third_party/systemlibs/protobuf.BUILD
index 504bc81a..4f105d33 100644
--- a/third_party/systemlibs/protobuf.BUILD
+++ b/third_party/systemlibs/protobuf.BUILD
@@ -65,12 +65,26 @@ cc_library(
     visibility = ["//visibility:public"],
 )
 
+cc_library(
+    name = "protobuf_lite",
+    linkopts = ["-lprotobuf-lite"],
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "protobuf_headers",
     linkopts = ["-lprotobuf"],
     visibility = ["//visibility:public"],
 )
 
+proto_lang_toolchain(
+    name = "cc_toolchain",
+    command_line = "--cpp_out=$(OUT)",
+    blacklisted_protos = [proto + "_proto" for proto in WELL_KNOWN_PROTO_MAP.keys()],
+    runtime = ":protobuf",
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "protoc_lib",
     linkopts = ["-lprotoc"],
