From f98ad31ffbb83ba849a8a97592c8c6f78e6f7fd0 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 13 May 2024 23:36:48 +1100
Subject: [PATCH 23/27] add kernel_timeout_internal target

---
 .../absl/system.absl.synchronization.BUILD        | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/third_party/absl/system.absl.synchronization.BUILD b/third_party/absl/system.absl.synchronization.BUILD
index c0fa37aa..b47309de 100644
--- a/third_party/absl/system.absl.synchronization.BUILD
+++ b/third_party/absl/system.absl.synchronization.BUILD
@@ -16,14 +16,29 @@ cc_library(
     ],
 )
 
+cc_library(
+    name = "kernel_timeout_internal",
+    visibility = [
+    ],
+    deps = [
+        "//absl/base",
+        "//absl/base:config",
+        "//absl/base:core_headers",
+        "//absl/base:raw_logging_internal",
+        "//absl/time",
+    ],
+)
+
 cc_library(
     name = "synchronization",
     linkopts = [
         "-labsl_synchronization",
+        "-labsl_kernel_timeout_internal",
         "-pthread",
     ],
     deps = [
         ":graphcycles_internal",
+        ":kernel_timeout_internal",
         "//absl/base",
         "//absl/base:atomic_hook",
         "//absl/base:dynamic_annotations",
