From 6712dade3b6ddd4881b5b7fd2150cf09eee40786 Mon Sep 17 00:00:00 2001
From: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
Date: Sat, 23 Dec 2023 00:59:32 -0500
Subject: [PATCH] Fix protobuf errors when using system protobuf

---
 tensorflow/api_template.__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tensorflow/api_template.__init__.py b/tensorflow/api_template.__init__.py
index 1ccf2fe07f0af..1573f5131bc9f 100644
--- a/tensorflow/api_template.__init__.py
+++ b/tensorflow/api_template.__init__.py
@@ -35,6 +35,8 @@
 import site as _site
 import sys as _sys
 
+# Do not remove this line; See https://github.com/tensorflow/tensorflow/issues/42596
+from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
 from tensorflow.python.tools import module_util as _module_util
 from tensorflow.python.util.lazy_loader import KerasLazyLoader as _KerasLazyLoader
