diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index d93e0df..6dc0f59 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -1968,6 +1968,7 @@ def tf_generate_proto_text_sources(name, srcs_relative_dir, srcs, protodeps = []
         outs = out_hdrs + out_srcs,
         visibility = visibility,
         cmd =
+            "LD_LIBRARY_PATH=$PREFIX/lib " +
             "$(location //tensorflow/tools/proto_text:gen_proto_text_functions) " +
             "$(@D) " + srcs_relative_dir + " $(SRCS)",
         tools = [
